pkg/driver/agent.go

```diff
@@ -341,20 +341,18 @@ func (ns *node) NodeExpandVolume(
 		)
 	}
 
-	// find if it is block device so that we don't attempt filesystem resize
-	st, err := os.Stat(req.GetVolumePath())
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to stat mountpath %s", err.Error())
-	}
+	isBlockMode := req.GetVolumeCapability().GetBlock() != nil
```

Comment:
volume capabilites are not passed in case of raw block volume, that was the reason it was done to check the mount path. Can you verify it?

Reply:
https://github.com/openebs/lvm-localpv/pull/85#issuecomment-854628262 Verified

Reply:
can you share the request structure from the log for raw block volume and for ext4 volume?

Reply:
it is the issue with CSI, it does not do that, then how it is working for you?

Reply:
> can you share the request structure from the log for raw block volume and for ext4 volume?

In above [comment](https://github.com/openebs/lvm-localpv/pull/85#issuecomment-854628262) I have pasted the request structure for different modes... 

> xfs:
```sh
I0604 11:04:25.055196       1 grpc.go:72] GRPC call: /csi.v1.Node/NodeExpandVolume requests {"capacity_range":{"required_bytes":5368709120},"staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-3816d67d-f495-45c8-88f8-e08604a16c8b/globalmount","volume_capability":{"AccessType":{"Mount":{"fs_type":"xfs"}},"access_mode":{"mode":1}},"volume_id":"pvc-3816d67d-f495-45c8-88f8-e08604a16c8b","volume_path":"/var/lib/kubelet/pods/400d9ba8-1333-4889-b405-246fe486a9a6/volumes/kubernetes.io~csi/pvc-3816d67d-f495-45c8-88f8-e08604a16c8b/mount"}
I0604 11:04:25.949620       1 grpc.go:81] GRPC response: {"capacity_bytes":5368709120}
```

volumemode block:
```sh
I0604 11:13:03.838028       1 grpc.go:72] GRPC call: /csi.v1.Node/NodeExpandVolume requests {"capacity_range":{"required_bytes":5368709120},"staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/staging/pvc-51631709-609f-4fde-a215-f054fe79c478","volume_capability":{"AccessType":{"Block":{}},"access_mode":{"mode":1}},"volume_id":"pvc-51631709-609f-4fde-a215-f054fe79c478","volume_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/publish/pvc-51631709-609f-4fde-a215-f054fe79c478/c30ad950-a6a2-41cf-a14f-1e566f5ac512"}
I0604 11:13:04.309051       1 grpc.go:81] GRPC response: {"capacity_bytes":5368709120}
```

Reply:
@mittachaitu as you can see volume capability is not there for NodeExpand call. You can print in your code path and see Getblock will be nil for raw block volume.

Reply:
> May be "-r" take care of raw block volume.

We are not passing "-r" in case of raw block volume

Reply:
ok, saw that. which k8s version you are using?

Reply:
the issue of not passing volume capability is fixed in 1.19.